src/ds/d500/d500-auto-calibration.cpp

```diff
@@ -9,310 +9,423 @@
-
-            _state = _calib_engine->get_triggered_calibration_state();
-            if (!(_state == calibration_state::IDLE ||
-                _state == calibration_state::COMPLETE))
-            {
-                LOG_ERROR("Calibration State is not Idle nor Complete - pleare restart the device");
-                throw std::runtime_error("OCC triggerred when Calibration State is not Idle not Complete");
-            }
+            LOG_ERROR("Calibration State is not Idle nor Complete - please restart the device");
+            throw std::runtime_error("OCC triggered when Calibration State is not Idle not Complete");
         }
+    }
+        
+    if (_mode == calibration_mode::ABORT)
+    {
+        // calibration state to be IN_PROCESS
+        auto calib_ans = get_triggered_calibration_status();
         
-        if (_mode == calibration_mode::ABORT)
+        if (!(calib_ans.state == calibration_state::PROCESS))
         {
-            // calibration state to be IN_PROCESS
-            _calib_engine->update_triggered_calibration_status();
-            _state = _calib_engine->get_triggered_calibration_state();
-            if (!(_state == calibration_state::PROCESS))
-            {
-                LOG_ERROR("Calibration State is not In Process - so it could not be aborted");
-                throw std::runtime_error("OCC aborted when Calibration State is not In Process");
-            }
+            LOG_ERROR("Calibration State is not In Process - so it could not be aborted");
+            throw std::runtime_error("OCC aborted when Calibration State is not In Process");
         }
     }
+}
 
-    void d500_auto_calibrated::get_mode_from_json(const std::string& json)
-    {
-        _mode = calibration_mode::RUN;
-        if (json.find("dry run") != std::string::npos)
-            _mode = calibration_mode::DRY_RUN;
-        else if (json.find("abort") != std::string::npos)
-            _mode = calibration_mode::ABORT;
-    }
+void d500_auto_calibrated::get_mode_from_json(const std::string& json)
+{
+    _mode = calibration_mode::RUN;
+    if (json.find("dry run") != std::string::npos)
+        _mode = calibration_mode::DRY_RUN;
+    else if (json.find("abort") != std::string::npos)
+        _mode = calibration_mode::ABORT;
+}
 
-    std::vector<uint8_t> d500_auto_calibrated::run_on_chip_calibration(int timeout_ms, std::string json, 
-        float* const health, rs2_update_progress_callback_sptr progress_callback)
+std::vector<uint8_t> d500_auto_calibrated::run_on_chip_calibration(int timeout_ms, std::string json, 
+    float* const health, rs2_update_progress_callback_sptr progress_callback)
+{
+    std::vector<uint8_t> res;
+    try
     {
-        std::vector<uint8_t> res;
-        try
+        get_mode_from_json(json);
+
+        // checking preconditions
+        check_preconditions_and_set_state();
+
+        // sending command to start calibration
+        _calib_engine->set(ds::SET_CALIB_MODE, static_cast<uint32_t>(_mode),
```

Comment:
It would be cleaner to wrap this in a `start_triggered_calibration` function

Reply:
ok